src/BPEssentials/CooldownHandler.cs

```diff
@@ -101,5 +100,28 @@ public int GetCooldown(string ID, string type, string key)
             if (!Cooldowns.ContainsKey(ID) || !Cooldowns[ID].ContainsKey(type) || !Cooldowns[ID][type].ContainsKey(key)) { return 0; }
             return Cooldowns[ID][type][key];
         }
+
+        private IEnumerator MethodTimer(int time, Action action, int repeat = -1)
+        {
+            int i = 0;
+            Core.Instance.Logger.Log("Starting cool down loop");
+            while (i < repeat && repeat != -1)
+            {
+                Core.Instance.Logger.Log("Starting cool down method");
+                action.Invoke();
+                Core.Instance.Logger.Log("Waiting " + time + "s");
+                yield return new WaitForSecondsRealtime(time);
+                if (repeat != -1)
+                {
+                    i++;
+                }
+            }
```

Comment:
What is this weird loop? Why not just use a for loop? Also, we don't need to log it each time.

Reply:
I rather i not be added 1 if there is no need... that's why the while loop (Ik its kinda weird but it seems like the best method to do it without duplicating the code

Reply:
A for loop would look better in this case. Please, change this to a for loop. Also, infinite loops without a break point are a code smell. If possible, change it so it can be stopped, with something like a [CancellationToken](https://docs.microsoft.com/en-us/dotnet/api/system.threading.cancellationtoken?view=netcore-3.1). Along with that, by changing it to a `for` loop you will now add `1` to `i` each loop, which is unneeded. Instead, check if `repeat` is smaller than `0`, and if so, start a `while` loop instead of the for loop.
```suggestion
            Core.Instance.Logger.Log("Starting cool down loop");
            for (var i = 0; i < repeat && repeat != -1; i++)
            {
                action.Invoke();
                yield return new WaitForSecondsRealtime(time);
            }
```

Reply:
I dont quit understand how you want to do the infinite loop for the save